Cycle_Sort/Cycle_Sort.java

```diff
@@ -0,0 +1,52 @@
+import java.io.*; 
+import java.util.*; 
+public class Cycle_Sort   
+{  
+static void sort(int a[], int n)  
+{  
+    int writes = 0,start,element,pos,temp,i;  
+    for (start = 0; start <= n - 2; start++) 
+    {  
+    element = a[start];  
+    pos = start;  
+    for (i = start + 1; i<n; i++)  
+        if (a[i] <element)  
+            pos++;  
+    if (pos == start)  
+        continue;  
+    while (element == a[pos])  
+        pos += 1;  
+    if (pos != start) {  
+    temp = element;  
+    element = a[pos];  
+        a[pos] = temp;    
+        writes++;  
+    }  
+    while (pos != start) {  
+        pos = start;  
+        for (i = start + 1; i<n; i++)  
+            if (a[i] <element)  
+                pos += 1;  
+        while (element == a[pos])  
+            pos += 1;  
+        if (element != a[pos]) {  
+          temp = element;  
+      element = a[pos];  
+          a[pos] = temp;    
+            writes++;  
+        }  
+    }  
+}  
+
```

Comment:
Please give new line between classes and functions.

Reply:
Also, there are no comments throughout, please add comments explaining the logic and a brief introduction about algorithm on top. Anyone who views the code should be able to learn from it.

Reply:
All Done

Reply:
Hi Disha, can you please fix a few more indentation errors, and try to change so many single line comments to multi line comment, then it will be ready for merge.